packages/decentraland-ecs/src/cli/setupUtils.ts

```diff
@@ -19,64 +21,112 @@ export async function copyDir(src: string, dest: string) {
   }
 }
 
+export const defaultHashMaker = (str: string) => 'b64-' + Buffer.from(str).toString('base64')
+
+export const getFilesFromFolder = ({
+  folder,
+  addOriginalPath,
+  ignorePattern,
+  customHashMaker,
+  rootFolder
+}: {
+  folder: string
+  addOriginalPath?: boolean
+  ignorePattern?: string
+  customHashMaker?: (str: string) => string
+  rootFolder?: string
+}) => {
+  const hashMaker = customHashMaker ? customHashMaker : defaultHashMaker
+
+  const allFiles = globSync('**/*', {
+    cwd: folder,
+    dot: false,
+    absolute: true
+  })
+    .map((file) => {
+      try {
+        if (!fs.statSync(file).isFile()) return
+      } catch (err) {
+        return
+      }
+      const _folder = folder.replace(/\\/gi, '/')
+      const key = file.replace(_folder, '').replace(/^\/+/, '')
+      return key
+    })
+    .filter(($) => !!$) as string[]
+
+  const ensureIgnorePattern = ignorePattern && ignorePattern !== '' ? ignorePattern : defaultDclIgnore()
+  const ig = ignore().add(ensureIgnorePattern)
+  const filteredFiles = ig.filter(allFiles)
+
+  return filteredFiles
+    .map((file) => {
+      const absolutePath = path.resolve(folder, file)
+      try {
+        if (!fs.statSync(absolutePath).isFile()) return
+      } catch (err) {
+        console.log(err)
+        return
+      }
+
+      const absoluteFolder = rootFolder ? rootFolder.replace(/\\/gi, '/') : folder.replace(/\\/gi, '/')
+      const relativeFilePathToRootFolder = absolutePath.replace(absoluteFolder, '').replace(/^\/+/, '')
+      const relativeFilePathToFolder = file.replace(absoluteFolder, '').replace(/^\/+/, '')
+
+      return {
+        file: relativeFilePathToFolder.toLowerCase(),
+        original_path: addOriginalPath ? absolutePath : undefined,
+        hash: hashMaker(relativeFilePathToRootFolder)
+      }
+    })
+    .filter(($) => !!$)
+}
+
 export function entityV3FromFolder({
   folder,
   addOriginalPath,
   ignorePattern,
-  customHashMaker
+  customHashMaker,
+  catalystRootFolder
 }: {
   folder: string
   addOriginalPath?: boolean
   ignorePattern?: string
   customHashMaker?: (str: string) => string
+  catalystRootFolder: string
 }) {
   const sceneJsonPath = path.resolve(folder, './scene.json')
-  const defaultHashMaker = (str: string) => 'b64-' + Buffer.from(str).toString('base64')
+  let isParcelScene = true
+
+  const assetJsonPath = path.resolve(folder, './asset.json')
```

Comment:
this could be a fn that helps us in the future.
```ts
function getProjectType() {
// lookFor assetJson
}
const isParcelScene = getProjectType() !== PORTABLE_EXPERIENCE
// or something like that
```